engine/src/flutter/lib/gpu/shader_library.cc

```diff
@@ -385,3 +385,22 @@ Dart_Handle InternalFlutterGpu_ShaderLibrary_GetShader(
   }
   return tonic::ToDart(shader.get());
 }
+
+Dart_Handle InternalFlutterGpu_ShaderLibrary_InitializeWithBytes(
+    Dart_Handle wrapper,
+    uint32_t bufferLength,
+    uint8_t* buffer) {
+  std::optional<std::string> out_error;
+  auto impeller_context = flutter::gpu::Context::GetDefaultContext(out_error);
+  if (out_error.has_value()) {
+    return tonic::ToDart(out_error.value());
+  }
+
+  auto mapping = std::make_unique<fml::NonOwnedMapping>(buffer, bufferLength);
```

Comment:
A NonOwnedMapping without a release callback will make the library reference data that can be deleted at any point and will lead to a dangling reference (presumably till the VM collects the bytedata). There must be a copy here.